### ISSUE TITLE
Add / fix support for text read mode

### ISSUE THREAD

**Issue by nick-hahner** [python-lz4/python-lz4#122]
```
In [17]: lz4.frame.open(input_file, mode='rt', newline='', encoding='utf-8')
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-17-ca68d3c17eba> in <module>()
----> 1 lz4.frame.open(input_file, mode='rt', newline='', encoding='utf-8')

/usr/local/lib/python3.6/site-packages/lz4/frame/__init__.py in open(filename, mode, encoding, errors, newline, block_size, block_linked, compression_level, content_checksum, block_checksum, auto_flush, return_bytearray, source_size)
    842         block_checksum=block_checksum,
    843         auto_flush=auto_flush,
--> 844         return_bytearray=return_bytearray,
    845     )
    846

/usr/local/lib/python3.6/site-packages/lz4/frame/__init__.py in __init__(self, filename, mode, block_size, block_linked, compression_level, content_checksum, block_checksum, auto_flush, return_bytearray, source_size)
    513             self._pos = 0
    514         else:
--> 515             raise ValueError('Invalid mode: {!r}'.format(mode))
    516
    517         if sys.version_info > (3, 6):

ValueError: Invalid mode: 'rt'
```

Looking at the source, it looks like mode is passed from https://github.com/python-lz4/python-lz4/blob/603c112a2c8dbaaa706780a94003aeb6888e36bb/lz4/frame/__init__.py#L837 to https://github.com/python-lz4/python-lz4/blob/603c112a2c8dbaaa706780a94003aeb6888e36bb/lz4/frame/__init__.py#L500

open() is happy with a mode of 'rt' and it passes it along unchanged to the LZ4FrameFile class which does not like 'rt'. Perhaps the mode should be changed to 'rb' when passing to the __init__ method, or perhaps the __init__ method should be happy with 'rt'.

Either way it breaks.

**Comment by jonathanunderwood** [comment]
Thanks for the report. Have checked a fix in to the text_io branch that I think addresses the issue, and will cut a new release shortly. If you had time to test the fix for your use case, that would be much appreciated.

**Comment by jonathanunderwood** [comment]
Fix is in version 1.0.1. Will close this - please re-open if you still see problems.
